libsrc/leddevice/dev_net/ProviderRestApi.cpp

```diff
@@ -0,0 +1,242 @@
+// Local-Hyperion includes
+#include "ProviderRestApi.h"
+
+// Qt includes
+#include <QEventLoop>
+#include <QNetworkReply>
+#include <QByteArray>
+
+//std includes
+#include <iostream>
+
+static const QChar ONE_SLASH = '/';
+
+ProviderRestApi::ProviderRestApi(const QString &host, const int &port, const QString &basePath)
+	:_log(Logger::getInstance("LDEDEVICE"))
+	  ,_networkManager(nullptr)
+	  ,_scheme("http")
+	  ,_hostname(host)
+	  ,_port(port)
+{
+	_networkManager = new QNetworkAccessManager();
+
+	_apiUrl.setScheme(_scheme);
+	_apiUrl.setHost(host);
+	_apiUrl.setPort(port);
+	_basePath = basePath;
+}
+
+ProviderRestApi::ProviderRestApi(const QString &host, const int &port)
+	: ProviderRestApi(host, port, "") {}
+
+ProviderRestApi::ProviderRestApi()
+	: ProviderRestApi("", -1) {}
+
+ProviderRestApi::~ProviderRestApi()
+{
+	if ( _networkManager != nullptr )
+	{
+		_networkManager->deleteLater();
+	}
+}
+
+void ProviderRestApi::setBasePath(const QString &basePath)
+{
+	_basePath.clear();
+	appendPath (_basePath, basePath );
+}
+
+void ProviderRestApi::setPath ( const QString &path )
+{
+	_path.clear();
+	appendPath (_path, path );
+}
+
+void ProviderRestApi::appendPath ( const QString &path )
+{
+	appendPath (_path, path );
+}
+
+void ProviderRestApi::appendPath (QString& path, const QString &appendPath)
+{
+	if ( !appendPath.isEmpty() && appendPath != ONE_SLASH )
+	{
+		if (path.isEmpty() || path == ONE_SLASH )
+		{
+			path.clear();
+			if (appendPath[0] != ONE_SLASH )
+			{
+				path.push_back(ONE_SLASH);
+			}
+		}
+		else if (path[path.size()-1] == ONE_SLASH && appendPath[0] == ONE_SLASH)
+		{
+			path.chop(1);
+		}
+		else if (path[path.size()-1] != ONE_SLASH && appendPath[0] != ONE_SLASH)
+		{
+			path.push_back(ONE_SLASH);
+		}
+		else
+		{
+			// Only one slash.
+		}
+
+		path.append(appendPath);
+	}
+}
+
+void ProviderRestApi::setFragment(const QString &fragment)
+{
+	_fragment = fragment;
+}
+
+void ProviderRestApi::setQuery(const QUrlQuery &query)
+{
+	_query = query;
+}
+
+QUrl ProviderRestApi::getUrl()
```

Comment:
it can be a ````const```` method

Reply:
Update Done.

Thanks for all the helpful feedback!!!